src/main/io/osd.c

```diff
@@ -395,7 +395,12 @@ static void osdDrawSingleElement(uint8_t item)
     case OSD_HOME_DIR:
         if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME)) {
             if (GPS_distanceToHome > 0) {
-                const int h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(attitude.values.yaw);
+                int h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(attitude.values.yaw);
+                // If MAG_Heading is not available, yaw starts with zero in a random direction. With GPS course over ground this can be roughly corrected
+                if (!sensors(SENSOR_MAG)) {
+                    h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(gpsSol.groundCourse);
```

Comment:
This is still the wrong place to do this - it should be done in https://github.com/betaflight/betaflight/blob/a618de9e8e59611ae0a484ea244ec6630d7411f7/src/main/flight/imu.c#L431 (where it is already done). Having this done in different ways in 2 different places (and one of them inconsistent as to where the heading is used (OSD is different to everything else in your approach) makes this hard to maintain and prone to breaking.
As a suggestion, instead of this change, introduce a `force_gps_heading` parameter, and have the IMU use GPS for heading when either the model is fixed wing, or `force_gps_heading` is enabled. For most pilots, having this switch to using a potentially misleading GPS heading if they have GPS installed is not how it should be done. If they have to first enable this in their configuration then at least they know that the 'home direction' indicator will become useless as soon as they don't fly straight forward.

Reply:
Thanks, but that's likely to complicated for me.
After finishing all that incremental tidy work, i think about stopping this PR...
Maybe i'll come back to you to know how to leave minimal traces...

Reply:
I think giving up on this would be a shame. I think we are already there. the change in behaviour that you are proposing can probably be implemented by changing `STATE(FIXED_WING)` in https://github.com/betaflight/betaflight/blob/a618de9e8e59611ae0a484ea244ec6630d7411f7/src/main/flight/imu.c#L429 to `STATE(FIXED_WING) || imuConfig()->force_gps_heading`. You can probably test that by hard coding it to true for testing.

At a pinch, the same (or similar) could be done for the altitude in https://github.com/betaflight/betaflight/blob/a618de9e8e59611ae0a484ea244ec6630d7411f7/src/main/flight/altitude.c#L209 by overriding the altitude estimate with the GPS altitude if the user sets a parameter to do so.